maps/kleibkhar/kleibkhar_turf.dm

```diff
@@ -73,5 +117,98 @@
 		footstep_type = /decl/footsteps/asteroid
 		color = null
 
+//
+//Terraforming
+//
+
+//Mud working
+var/global/list/exterior_mud_dark_radial_choices
+/turf/exterior/mud/dark/proc/cache_radial_shovel_interactions()
+		var/image/radial_choice_tile    = image('icons/obj/hydroponics/hydroponics_machines.dmi', null, "soil",)
+		var/image/radial_choice_extract = image('icons/obj/items/tool/shovels/shovel.dmi', ICON_STATE_WORLD)
+		var/image/radial_choice_dig     = image('icons/obj/structures/pit.dmi', "pit1")
+		radial_choice_tile.name    = "Tile Soil"
+		radial_choice_extract.name = "Extract Clay"
+		radial_choice_dig.name     = "Dig Pit"
+		exterior_mud_dark_radial_choices = list("Tile Soil" = radial_choice_tile, "Extract Clay" = radial_choice_extract, "Dig Pit" = radial_choice_dig)
+
+/turf/exterior/mud/dark/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/minihoe))
+		if(user.do_skilled(8 SECONDS, SKILL_HAULING, src) && !QDELETED(W))
+			to_chat(user, SPAN_NOTICE("You remove the muddy layer of the ground."))
+			ChangeTurf(/turf/exterior/barren)
+		return TRUE
+
+	if(IS_SHOVEL(W))
+		if((locate(/obj/machinery/portable_atmospherics/hydroponics/soil) in contents))
```

Comment:
Out of curiosity, are the additional parentheses here necessary?

Reply:
Nope, probably a leftover.